### PR TITLE
fix(locksmith): lock managers and key owners can generate QR code

### PR DESCRIPTION
**Pull request by julien51**
# Description

A key owner and the lock manager should both be able to generate a QR code.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/src/routes/v2/ticket.ts [hunk 1]
```diff
@@ -5,13 +5,11 @@ import {
...
 import { keyOwnerMiddleware } from '../../utils/middlewares/keyOwnerMiddleware'
-import {
-  authenticatedMiddleware,
-  applicationOnlyMiddleware,
-} from '../../utils/middlewares/auth'
+import { authenticatedMiddleware } from '../../utils/middlewares/auth'
 import { isVerifierMiddleware } from '../../utils/middlewares/isVerifierMiddleware'
 import { Web3Service } from '@unlock-protocol/unlock-js'
 import { lockManagerMiddleware } from './../../utils/middlewares/lockManager'
+import { lockManagerOrKeyOwnerMiddleware } from '../../utils/middlewares/lockManagerOrKeyOwner'
```
Comment (searchableguy) [inline locksmith/src/routes/v2/ticket.ts:12]: Need to add file to git?

**Review by searchableguy — CHANGES_REQUESTED** [review]

File: locksmith/src/routes/v2/ticket.ts (hunk 1 above)
Comment (julien51) [reply locksmith/src/routes/v2/ticket.ts:12]: yes :D

File: locksmith/src/utils/middlewares/lockManagerOrKeyOwner.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import Normalizer from '../normalizer'
+
+export const lockManagerOrKeyOwnerMiddleware: RequestHandler = async (
+  req,
+  res,
+  next
+) => {
+  const lockAddress = Normalizer.ethereumAddress(req.params.lockAddress)
+  const network = Number(req.params.network)
+  const userAddress = Normalizer.ethereumAddress(req.user!.walletAddress!)
```
Comment (searchableguy) [inline locksmith/src/utils/middlewares/lockManagerOrKeyOwner.ts:13]: Normalizing isn't required. We already do it for user address.

File: locksmith/src/utils/middlewares/lockManagerOrKeyOwner.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  const isLockManager = await web3Service.isLockManager(
+    lockAddress,
+    userAddress,
+    network
+  )
+
+  const keyOwner = Normalizer.ethereumAddress(
+    await web3Service.ownerOf(lockAddress, tokenId, network)
+  )
+
```
Comment (searchableguy) [inline locksmith/src/utils/middlewares/lockManagerOrKeyOwner.ts:39]: Batch please.

**Review by searchableguy — APPROVED** [review]

File: locksmith/src/utils/middlewares/lockManagerOrKeyOwner.ts (hunk 2 above)
Comment (julien51) [inline locksmith/src/utils/middlewares/lockManagerOrKeyOwner.ts:13]:
```suggestion
  const userAddress = req.user!.walletAddress
```
Good point! Thanks

File: locksmith/src/utils/middlewares/lockManagerOrKeyOwner.ts (hunk 3 above)
Comment (julien51) [reply locksmith/src/utils/middlewares/lockManagerOrKeyOwner.ts]: Fixed!

**Pull request merged**